api/telemetry/v1alpha1/collector_types.go

```diff
@@ -47,6 +47,14 @@ type MemoryLimiter struct {
 	MemorySpikePercentage uint32 `json:"spike_limit_percentage"`
 }
 
+// Configuration for persistence, will be used to generate
+// the filestorage extension.
+type PersistenceConfig struct {
+	// The directory where logs will be persisted, if not set, or invalid
+	// defaults will be used, based on the OS.
```

Comment:
I would rewrite this documentation comment to something like this:
```suggestion
	// The directory where logs will be persisted.
	// If unset or invalid path is given, then OS specific default values will be used.
```